Source/WebCore/css/CSSCounterStyleDescriptors.cpp

```diff
@@ -501,36 +650,131 @@ String CSSCounterStyleDescriptors::padCSSText() const
     return m_pad.cssText();
 }
 
+void CSSCounterStyleDescriptors::padCSSText(StringBuilder& builder) const
+{
+    if (!m_explicitlySetDescriptors.contains(ExplicitlySetDescriptors::Pad))
+        return;
+    m_pad.cssText(builder);
+}
+
+// MARK: - Fallback
+
+template<typename Maker> static decltype(auto) serializeFallback(const CSSCounterStyleDescriptors::Name& fallbackName, Maker&& maker)
+{
+    return maker(fallbackName);
+}
+
+bool CSSCounterStyleDescriptors::hasFallbackCSSText() const
+{
+    if (!m_explicitlySetDescriptors.contains(ExplicitlySetDescriptors::Fallback))
+        return false;
+    return !m_fallbackName.isEmpty();
+}
+
 String CSSCounterStyleDescriptors::fallbackCSSText() const
 {
     if (!m_explicitlySetDescriptors.contains(ExplicitlySetDescriptors::Fallback))
         return emptyString();
-    return makeString(m_fallbackName);
+    return serializeFallback(m_fallbackName, SerializeUsingMakeString { });
+}
+
+void CSSCounterStyleDescriptors::fallbackCSSText(StringBuilder& builder) const
+{
+    if (!m_explicitlySetDescriptors.contains(ExplicitlySetDescriptors::Fallback))
+        return;
+    serializeFallback(m_fallbackName, SerializeUsingStringBuilder { builder });
+}
+
+// MARK: - Symbols
+
+bool CSSCounterStyleDescriptors::Symbol::hasCSSText() const
+{
+    if (isCustomIdent)
+        return !text.isEmpty();
+
+    // If the symbol is not a custom ident, it is serialized as a string, which
+    // will always at least include quotation marks, even for empty strings.
+    return true;
```

Comment:
This comment is so clear, and given that, the code I would expect is:
```
return !(isCustomIdent && text.isEmpty());
```
I guess I’m a fan of boolean expressions when they can be readable.